server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -117,6 +117,20 @@ void recoverFromStore(final IndexShard indexShard, ActionListener<Boolean> liste
         }
     }
 
+    void recoverFromRemoteStore(final IndexShard indexShard, ActionListener<Boolean> listener) {
+        if (canRecover(indexShard)) {
+            RecoverySource.Type recoveryType = indexShard.recoveryState().getRecoverySource().getType();
+            assert recoveryType == RecoverySource.Type.REMOTE_STORE : "expected remote store recovery type but was: " + recoveryType;
+            ActionListener.completeWith(recoveryListener(indexShard, listener), () -> {
+                logger.debug("starting recovery from remote store ...");
+                recoverFromRemoteStore(indexShard);
+                return true;
+            });
+        } else {
+            listener.onResponse(false);
+        }
+    }
+
```

Comment:
Can we extend `StoreRecovery` to `RemoteStoreRecovery` and override `recoverFromStore`

Reply:
`StoreRecovery` currently contains: `recoverFromStore`, `recoverFromRepository` and `recoverFromLocalShards` where the source of recovery is different for each of these methods. That is why I added one more method where recovery source is remote store.